src/test/e2e/api/admin/segment.e2e.test.ts

```diff
@@ -108,14 +108,18 @@ const validateSegment = (
 
 beforeAll(async () => {
     db = await dbInit('segments_api_serial', getLogger);
-    app = await setupAppWithCustomConfig(db.stores, {
-        experimental: {
-            flags: {
-                strictSchemaValidation: true,
-                anonymiseEventLog: true,
+    app = await setupAppWithCustomConfig(
+        db.stores,
+        {
+            experimental: {
+                flags: {
+                    anonymiseEventLog: true,
+                    detectSegmentUsageInChangeRequests: true,
+                },
             },
         },
-    });
+        db.rawDatabase,
```

Comment:
We didn't pass the database to the setup before. This meant that we never hit the real services underneath, but used the fake ones instead. For the segment usage read model, that was a problem, so I hooked it up.